sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/AsynchronousMetricStorage.java

```diff
@@ -194,14 +202,41 @@ public void record(Attributes attributes, T accumulation) {
                 + ").");
         return;
       }
-      // TODO: error on metric overwrites
+
+      // Check there is not already a recording for the attributes
```

Comment:
I can't find making this an error in the spec

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-updowncounter-creation

`User code is recommended not to provide more than one Measurement with the same attributes in a single callback. If it happens, the SDK MAY decide how to handle it. For example, during the callback invocation if two measurements value=1, attributes={pid:4, bitness:64} and value=2, attributes={pid:4, bitness:64} are reported, the SDK can decide to simply let them pass through (so the downstream consumer can handle duplication), drop the entire data, pick the last one, or something else.`

I personally am not a fan of dropping (strongly) or logging (less strongly) for this, it's similar to how `Span.setAttribute` can be called with the same key multiple times, rarely needed but seems overstepping by treating as an error.

Reply:
That's the line that I was going off of. The current behavior is "last recording wins", and this PR changes it to "first recording wins, and log". 

I don't have a strong preference for whether first or last wins, but I think that logging (at some level) is important because without logging it could be difficult to diagnose.